--- origsrc/ggz-utils-0.0.14.1/metaserv/metaserv/metaserv.c	2008-01-05 13:03:23.000000000 -0600
+++ src/ggz-utils-0.0.14.1/metaserv/metaserv/metaserv.c	2008-06-06 13:43:27.515625000 -0500
@@ -1004,22 +1004,18 @@
 	int fd;
 	struct sockaddr_in peername;
 	socklen_t namelen;
-	int ret;
-	struct hostent host, *hp;
-	int herr;
+	struct hostent *host;
 	const char *peer;
 	static long session = 0;
 	int tmpsession;
-	char tmp[1024];
 
 	fd = *(int*)arg;
 	free(arg);
 
 	namelen = sizeof(struct sockaddr_in);
 	getpeername(fd, (struct sockaddr*)&peername, &namelen);
-	ret = gethostbyaddr_r(&peername.sin_addr, sizeof(struct in_addr), AF_INET,
-		&host, tmp, sizeof(tmp), &hp, &herr);
-	if(!ret) peer = host.h_name;
+	host = gethostbyaddr((char *)&peername.sin_addr, sizeof(struct in_addr), AF_INET);
+	if(host) peer = host->h_name;
 	else peer = "(unknown)";
 
 	tmpsession = session++;
